csrc/core/determinism.hpp

```diff
@@ -0,0 +1,33 @@
+#pragma once
```

Comment:
Is there any way for us to avoid heavily hacking on envvars? Instead we can do something like pass in `batch_invariant=true` as a kernel's input? Not sure whether this is actually possible so will leave this comment as a suggestion here.

Reply:
this will involve a lot of plumbing and I think we may just want to use env hacks to toggle it for the time being